lib/attr_encrypted.rb

```diff
@@ -303,12 +338,23 @@ def encrypt(attribute, value)
 
       # Returns attr_encrypted options evaluated in the current object's scope for the attribute specified
       def evaluated_attr_encrypted_options_for(attribute)
-        if evaluate_attr_encrypted_option(self.class.encrypted_attributes[attribute.to_sym][:mode]) == :per_attribute_iv_and_salt
-          load_iv_for_attribute(attribute, self.class.encrypted_attributes[attribute.to_sym][:algorithm])
-          load_salt_for_attribute(attribute)
+        evaluated_options = Hash.new
+        attribute_option_value = self.class.encrypted_attributes[attribute.to_sym][:attribute]
+        self.class.encrypted_attributes[attribute.to_sym].map do |option, value|
+          evaluated_options[option] = evaluate_attr_encrypted_option(value)
         end
 
-        self.class.encrypted_attributes[attribute.to_sym].inject({}) { |hash, (option, value)| hash[option] = evaluate_attr_encrypted_option(value); hash }
+        evaluated_options[:attribute] = attribute_option_value
```

Comment:
Can `evaluated_options` above just be initialized with this? I don't think `attribute_option_value` is used anywhere else in this method. It would have to be moved before the "evaluate options" block though, does order matter?

```ruby
evaluated_options = { attribute: self.class.encrypted_attributes[attribute.to_sym][:attribute] }
```